src/selector.rs

```diff
@@ -13,35 +17,41 @@ use ElementRef;
 #[derive(Debug, Clone, PartialEq)]
 pub struct Selector {
     /// The CSS selectors.
-    pub selectors: Vec<parser::Selector<Simple>>,
+    pub selectors: SmallVec<[parser::Selector<Simple>; 1]>,
 }
 
 impl Selector {
     /// Parses a CSS selector group.
-    ///
-    /// No meaningful error can be returned here, due to a limitation of the `selectors` and
-    /// `cssparser` crates.
-    pub fn parse(selectors: &str) -> Result<Self, ()> {
-        let mut parser = cssparser::Parser::new(selectors);
+
+    pub fn parse<'t, 'i>(
+        selectors: &'i str,
+    ) -> Result<Self, cssparser::ParseError<'t, SelectorParseErrorKind<'i>>> {
```

Comment:
Should this not be the same lifetime in both places? I don't understand where `'t` comes from.

Reply:
I tried to [implement their new Selector](https://docs.rs/selectors/0.19.0/selectors/parser/struct.SelectorList.html#method.parse), now returning a useful error message (yay!) and requiring an additional lifetime.

Possibly `Result<Self, cssparser::ParseError<'i, SelectorParseErrorKind<'i>>>` would be better?